lips/lip-10.md

```diff
@@ -73,6 +75,32 @@ struct LibraIdDomain {
 * The Libra ID domain can be created as part of creating a `ParentVASP` account.
 * An entity (potentially the Association) may at some point choose to expose an open source application(*Indexer*) that will allow indexing available Libra IDs and could provide a convenient REST API for applications to fetch information about the domains. The API of such an indexer is out of scope of this RFC.
 
+## Libra ID domain events
+
+Blockchain contract that manages Libra ID domains must emit an event every time Libra ID domain is created, removed or updated. Those events are critical for applications to be able to efficiently index existing Libra ID domains.
+
+While Libra ID domains published into VASP accounts, Libra ID domain events are published under the Treasury Compliance account. We consolidate events under single account to allow indexers to follow single event stream.
+
+To store events, `LibraIdDomainManager` resource is published under the Treasure Compliance account(address `0xB1E55ED`).
+
+```
+resource struct LibraIdDomainManager {
+    /// Events are emmitted 
+    libra_id_domain_events: Event::EventHandle<Self::LibraIdDomainEvent>,
+}
+
+struct LibraIdDomainEvent {
+    removed: bool,
+    domain: LibraIdDomain,
```

Comment:
what do we benefit by having the actual value here? this seems to be a potential waste of resources.

Reply:
I am wondering about the same thing. It's likely that we need to add another field for old domain on update cases. In that case, maybe we do not reveal the values but just an event that tells people this address has an update for libraid domain. The trade-off is people will need to look it up by themselves.

Reply:
So there are basically two alternatives:
- Pass entire value like we do right now
- Pass some 'incremental update' struct that has array of (field, value)

Second option seem to be less than ideal as the contract will have to figure out how to generate it, and it is also not going to be strictly typed (e.g. what type `value` part of such incremental update should be?)

I think if we had large frequently updated structure I could understand argument of saving resources by not providing full value, but in current circumstances it seem pretty natural choice to just put 'latest value' here

Reply:
Another way is not to have any values (even incremental ones) in the event. The only message the event conveys is "this address has an update with libra id domains".

Reply:
What @andll has here is consistent with the style we use in other events. Elsewhere, we try to make events self-contained so you can understand them without needing to look at the associated state/transactions.

In this case, I think it makes sense to include the domain in the event. If you do that, the domain indexer will only need to look at the `LibraIdDomainEvent` stream to know the complete set of available domains + their owners.

Reply:
Yeah, I still think having value is much easier for indexer to process.